mountpoint-s3/src/run.rs

```diff
@@ -254,3 +272,51 @@ fn create_pid_file() -> anyhow::Result<()> {
     }
     Ok(())
 }
+
+pub struct UserSwitcher {
+    uid: Uid,
+    gid: Gid,
+}
+
+impl UserSwitcher {
+    pub fn new(username: &str) -> anyhow::Result<Self> {
+        let username = CString::new(username)?;
+
+        // Get user info
+        let passwd_ptr = unsafe { getpwnam(username.as_ptr()) };
```

Comment:
Why is this unsafe, and what makes us think this particular call is safe?

Reply:
Is this a better approach? https://docs.rs/nix/latest/nix/unistd/struct.User.html#method.from_name
Would it avoid usage of unsafe?